phoenix-core/src/it/java/org/apache/phoenix/end2end/SystemCatalogCreationOnConnectionIT.java

```diff
@@ -658,4 +670,30 @@ private Properties getClientProperties(boolean nsMappingEnabled, boolean systemT
         assertFalse(isSystemNamespaceCreated());
         return driver;
     }
+
+    @Test
+    public void testDoNotUpgradePropSet() throws Exception {
+        String tableName = "HBASE_SYNTH_TEST";
+        startMiniClusterWithToggleNamespaceMapping(Boolean.FALSE.toString());
+        Properties propsDoNotUpgradePropSet = new Properties();
+        // Create a dummy connection to make sure we have all system tables in place
+        Connection con1 = DriverManager.getConnection(getJdbcUrl(),propsDoNotUpgradePropSet);
+        String ddl = "CREATE TABLE "+ tableName +" (PK1 VARCHAR not null, PK2 VARCHAR not null, "
+                + "COL1 varchar, COL2 varchar " + "CONSTRAINT pk PRIMARY KEY(PK1,PK2))";
+        con1.createStatement().execute(ddl);
+        con1.createStatement().execute("UPSERT INTO " + tableName + " values ('pk1','pk2','c1','c2')");
+        con1.commit();
+        // Stop HMaster to check if we can create connection without active HMaster
+        testUtil.getMiniHBaseCluster().getMaster().stopMaster();
```

Comment:
Restart hmaster later?

Reply:
I tried to start hmaster later but it didn't work. I'll check and put this one in next commit.